yrs/src/types/text.rs

```diff
@@ -156,17 +156,23 @@ impl Text {
         mut attributes: Attrs,
     ) {
         if let Some(mut pos) = self.find_position(txn, index) {
-            pos.unset_missing(&mut attributes);
-            Text::minimize_attr_changes(&mut pos, &attributes);
-            let negated_attrs = self.insert_attributes(txn, &mut pos, attributes);
+            let negated_attrs = if !attributes.is_empty() {
+                pos.unset_missing(&mut attributes);
+                Text::minimize_attr_changes(&mut pos, &attributes);
+                Some(self.insert_attributes(txn, &mut pos, attributes))
+            } else {
+                None
+            };
```

Comment:
I want to raise a question I have about this code here. It seems to cause it to be hard to insert a string with no attributes into one with attributes.

Reply:
I'm not sure if this is something that should not be allowed, but the ```Diff``` struct defines ```Attrs``` as optional. So I think it might be worth a mention.
https://github.com/y-crdt/y-crdt/blob/96aa3d1b24754400fbd9013235e5edbd97fd22f0/yrs/src/types/text.rs#L695-L697

Reply:
Add a test:
```Rust
use yrs::{
    types::{text::Diff, Attrs},
    Doc,
};

#[test]
fn insert_string_with_no_attribute() {
    let doc = Doc::new();
    let mut txn = doc.transact();
    let txt = txn.get_text("test");

    let attrs = Attrs::from([("a".into(), "a".into())]);
    txt.insert_with_attributes(&mut txn, 0, "ac", attrs.clone());
    txt.insert_with_attributes(&mut txn, 1, "b", Attrs::new());

    let expect = vec![
        Diff::Insert("a".into(), Some(Box::new(attrs.clone()))),
        Diff::Insert("b".into(), None),
        Diff::Insert("c".into(), Some(Box::new(attrs.clone()))),
    ];

    assert!(txt.diff(&mut txn).eq(&expect))
}
```

Reply:
My bad here. I thought that this was a micro optimisation that doesn't interfere with library behaviour - hence no tests were broken. I put too much trust into our test suite. Thank you for noticing that.